pkg/api/types.go

```diff
@@ -2979,6 +2979,9 @@ type AttachedVolume struct {
 
 	// DevicePath represents the device path where the volume should be available
 	DevicePath string
+
+	// Has the process of attaching the volume finished already?
+	IsAttached bool
```

Comment:
The parent field name is unfortunate ("why are there unattached volumes in the volumesAttached list?")

This also means that existing API clients using this list to do things against attached volumes can no longer depend on items in this list actually being attached. That seems like a breaking change in behavior. I wonder if a parallel field with volumes pending attachment, or a replacement field with richer volume status would be better. 

You also need to update the v1 types.go file